### PR TITLE
Some blob fixes

### LINKED ISSUE #11149: A bunch of blob chems are absolutely broken
Sorium and darkmatter don't throw on expanding
Skin melter and boiling oil don't ignite
Probably a bunch more chems don't work right

### LINKED ISSUE #11208: Expanding onto a blobbernaut as blob attacks instead of expanding
Blobbernauts aren't immune to blob_act.

### PR DESCRIPTION
**Pull request by phil235**
Fixes blob sorium and dark matter vortex not working correctly. Fixes #11149
Fixes blob expansion attacking its own blob mobs (blobbernaut,etc). Fixes #11208

### CODE REVIEW

**Comment by Aranclanos** [comment]
we don't really use those loops

File: code/game/gamemodes/blob/powers.dm [hunk 1]
```diff
@@ -275,9 +275,13 @@
 		return
 	last_attack = world.time
 	OB.expand(T, 0, blob_reagent_datum.color)
-	for(var/mob/living/L in T)
-		blob_reagent_datum.reaction_mob(L, TOUCH, 25)
-		blob_reagent_datum.send_message(L)
+	mob_loop:
+		for(var/mob/living/L in T)
+			for(var/F in L.faction)
+				if(F == "blob") // no friendly fire
```
Comment (duncathan) [inline code/game/gamemodes/blob/powers.dm]: why not just if("blob" in L.faction)?
then you wouldn't need a named loop; there'd only be the one

**Comment by RemieRichards** [comment]
@Aranclanos there's a decent amount of named loops in our code, and there's nothing wrong with them. The only issue in this case is it's entirely unnecessary as @duncathan has pointed out.

File: code/game/gamemodes/blob/powers.dm (hunk 1 above)
Comment (phil235) [inline code/game/gamemodes/blob/powers.dm]: hmm yeah, I'll try that instead.

**Comment by phil235** [comment]
I'm going to move this PR's content to my other reagents reaction PR so I don't have conflicts to solve.

**Comment by Aranclanos** [comment]
named loop fangays git out

**Comment by Iamgoofball** [comment]
>using goto ever

**Pull request closed without merging**